### ISSUE TITLE
Issue with $repositoryname

### ISSUE THREAD

**Issue by sglienke** [ForkIssues/TrackerWin#1826]
It seems that the variable contains the parent folder name instead of the name of the respository I gave it (I assume you are using something like this: https://stackoverflow.com/a/15716016/587106)

I was trying to use this in a custom action to open the commit in the browser (using GitLab) and the url contains the parent folder name instead of the name I have for the repository in Fork.

**Comment by DanPristupov** [comment]
Yes, `$repositoryname` is a repository folder name.

**Comment by sglienke** [comment]
Is there a way to use the actual repository name so I can use this in the url of a custom action across multiple repositories?

**Comment by bergamin** [comment]
Hi, sorry to comment unrelated stuff in this issue but, where can I find documentation on these variables? That sure would help me setting up some stuff. Mainly the repository name if it actually showed the actual name of the repository and not the folder name I used when cloning

**Comment by DanPristupov** [comment]
@bergamin 

Right in the custom commands dialogs:

<img width="832" alt="Screenshot 2023-09-13 at 8 43 54" src="https://github.com/fork-dev/TrackerWin/assets/618115/eaa2f9f1-79fa-4f6d-832f-d27f4ee11de5">

<img width="694" alt="Screenshot 2023-09-13 at 8 44 21" src="https://github.com/fork-dev/TrackerWin/assets/618115/b8889b88-0a64-45c6-993e-95ddb0d40a62">

**Comment by bergamin** [comment]
@DanPristupov, ah I see. I thought this was related to this screen, but it seems not...

![image](https://github.com/fork-dev/TrackerWin/assets/14339291/c8ebb0f6-992b-4d2f-aeb5-b0f79164c3f1)

which would be awesome for resolving [this problem](https://github.com/fork-dev/TrackerWin/issues/1449#issuecomment-1039095403) I have.

I'll open a dedicated feature request about this. I thought I had done that back then. I don't know why I just added it as a comment in another issue...

I have to manage, as of today, 122 repositories. #1449 and #1706 + variables in the issue tracking linking thingy would be very helpful to have. Also, an option to solve #1826 would be to add the ability to create custom variables based on regexes. We could with that take the origin link and extract just the actual repo name.

**Comment by DanPristupov** [comment]
No, that's different. `${repo:name}` is the name of the local folder on disk. It doesn't make sense to pass it to issue tracker rules.

**Comment by bergamin** [comment]
@DanPristupov, fair enough. The variable name is misleading, though. It is not the repository name in the sense most people think of what the repository name is.

Still, we could solve the issue mentioned by @sglienke through allowing custom variables. This way you don't have to provide something git doesn't have (as per SO link in the original message) and it will allow us to extract the repo name from the origin link, which is something you can provide as a native variable.
